### DISCUSSION TITLE
Hide support text on gizchina.gr

### DISCUSSION THREAD

**Discussion by DimitrisKalaboukas** [uBlockOrigin/uAssets#22018] [Q&A] [answered]
`https://gr.gizchina.com/2024/01/16/android-12-tablet-me-8gb-ram-5g-mpataria-7000-mah-xmas-2/`

I want to block the annoying paragraph between “-----------”, which is added at the bottom of every article asking for support, but I haven’t managed to write an effective filter. Could anyone help?

https://github.com/uBlockOrigin/uAssets/assets/101707608/28b1f161-eeb0-4e6f-98c9-ab543cb5b0f1

**Comment by partingscientist** [comment]
Very hacky, but these should work.

```adblock
gizchina.com##+js(rmnt, #text, -----------)
gizchina.com##+js(rmnt, #text, συνεργατών, condition, συνδέσμους)
gizchina.com##+js(rmnt, #text, σύνδεσμο, condition, ποσοστό)
gizchina.com##+js(rmnt, #text, υποστήριξη, condition, ευχαριστούμε)
gizchina.com##+js(rmnt, b, Gizchina Greece, sedCount, 1)
gizchina.com##+js(rmnt, b, εδώ)
```

I haven't found any false positives after opening several articles so far, but do beware of them. The maintainers might come up with a better solution.

**Reply by garry-ut99** [reply]
> DimitrisKalaboukas : I want to block the annoying paragraph between

You need to use this : https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library#remove-node-textjs-


> partingscientist : I haven't found any false positives after opening several articles so far, but do beware of them. 

Maybe your filter will be just fine, but if you consider that it might be too risky, here is a more safe solution :

```
gr.gizchina.com##+js(rmnt, #text, /-----------|Κάποιες από τις αναρτήσεις μας μπορεί να περιέχουν συνδέσμους συνεργατών. Το|μπορεί να λαμβάνει ένα μικρό ποσοστό| εάν κάνετε κλικ σε έναν σύνδεσμο και αγοράσετε κάποιο προιόν. Αν θέλετε περισσότερες λεπτομέρειες|μπορείτε να μάθετε πώς χρησιμοποιούμε τους συνδέσμους συνεργατών./)
gr.gizchina.com##.post-content.clearfix > b:has-text(/Gizchina Greece|Σας ευχαριστούμε για την υποστήριξη!/)
gr.gizchina.com##.post-content.clearfix > b:has(> [href="https://gr.gizchina.com/disclaimer/"])
gr.gizchina.com##.post-content.clearfix > b:has(> [href="https://gr.gizchina.com/disclaimer/"]) + b
gr.gizchina.com##.post-content.clearfix > br
```

Your solution also requires this additional filter: `gr.gizchina.com##.post-content.clearfix > br` to hide `br` elements which still unnecessarily take visual space. (by the way, I haven't come here to fight which solution is better, really, as each of them has it's pros/cons, just to discuss and provide some alternative).

**Comment by D4niloMR** [comment]
You can use these two:

```adb
gr.gizchina.com##+js(rmnt, #text, /^(\s*-----------\s*|Κάποιες από τις αναρτήσεις μας μπορεί να περιέχουν συνδέσμους συνεργατών\. Το | μπορεί να λαμβάνει ένα μικρό ποσοστό\, εάν κάνετε κλικ σε έναν σύνδεσμο και αγοράσετε κάποιο προιόν\. Αν θέλετε περισσότερες λεπτομέρειες\, | μπορείτε να μάθετε πώς χρησιμοποιούμε τους συνδέσμους συνεργατών\. )$/)
gr.gizchina.com##.post-content > blockquote:last-of-type ~ :is(div[style^="font-size: 0px; height: 0px;"] + br + b, br + b + b, br + b + b + b, br:not(p + br))
```

Rest assured about breakages and legit content removed, as this filter is VERY specific.

**Reply by garry-ut99** [reply]
Works fine on Desktop, but on Mobile some elements not hidden.

**Reply by D4niloMR** [reply]
Yeah it's too specific, this will work on both

```adb
gr.gizchina.com##.post-content > :is(br + b, br + b + b, br + b + b + b, br:not(p:last-of-type + br))
```

**Comment by amelialily2584** [comment]
I have Same question but thanks it is solved already, It is very helpful plateform.

**Comment by DimitrisKalaboukas** [comment]
@D4niloMR The message has changed and is now partially blocked:

https://www.gizchina.gr/2024/10/29/windows-11-god-mode-how-t0-activate-it-and-why/

Could you help to adjust the rules?

**Reply by garry-ut99** [reply]
```adb
gizchina.gr##+js(rmnt, #text, /^(\s*-----------\s*|Μπορεί να λάβουμε ένα μικρό ποσοστό εάν κάνετε κλικ σε έναν σύνδεσμο και αγοράσετε κάποιο προϊόν\. Για περισσότερες λεπτομέρειες\, | μπορείτε να μάθετε πώς χρησιμοποιούμε τους συνδέσμους συνεργατών\. |Σας ευχαριστούμε για την υποστήριξη\.|Ακολουθήστε το | για νέα και ειδήσεις στον χώρο της τεχνολογίας\. Αν ψάχνετε HOT προσφορές και κουπόνια για κινητά και gadgets\, κάντε εγγραφή στο | στο )$/)
gizchina.gr##.post-content > :is(br + b, br + b + b, br + b + b + b, br:not(p:last-of-type + br))
```

**Reply by DimitrisKalaboukas** [reply]
@garry-ut99 Thank you very much for your response. But also a fragmented laste paragraph:

εδώ Ακολουθήστε το Gizchina Greece στο Google News για νέα και ειδήσεις στον χώρο της τεχνολογίας. Αν ψάχνετε HOT προσφορές και κουπόνια για κινητά και gadgets, κάντε εγγραφή στο στο

is left behind and should be blocked.

**Reply by garry-ut99** [reply]
I've updated the filter in the previous comment, should be fine now.

By the way: you haven't quoted "[κανάλι μας](https://t.me/gizchinagr)Telegram."  - I'm not sure whether because you've forgotten or because you didn't want it to be hidden, anyway I've hidden it too just in case, if you don't want to hide it, just remove ` br + b + b, br + b + b + b,` from the filter.

**Reply by DimitrisKalaboukas** [reply]
Well done!!
They were already blocked by the previous rules, that's why I didn't mentioned them and I said "fragmented paragraph".

**Reply by garry-ut99** [reply]
One more thing: it has turned out that only the scriptlet filter was required to be modified, and not the cosmetic filter. I modified the previous cosmetic filter unnecessarily (the upper one): https://github.com/uBlockOrigin/uAssets/discussions/22018#discussioncomment-8150934, because I overlooked another one (at the bottom) which still works just fine: https://github.com/uBlockOrigin/uAssets/discussions/22018#discussioncomment-8159617, which I've noticed only right now.

(by the way: DimitrisKalaboukas : "rules" - they are filters, not rules, as per https://github.com/uBlockOrigin/uAssets/discussions/24776#discussioncomment-10311649)

**Reply by krystian3w** [reply]
He may be blocked and if he replies it will be from his account "freezer2022".

**Comment by DimitrisKalaboukas** [comment]
@freezer2022 The scriptlet rule needed some small adjustments:

`gizchina.gr##+js(rmnt, #text, /^(\s*-----------\s*|Μπορεί να λάβουμε ένα μικρό ποσοστό εάν κάνετε κλικ σε έναν σύνδεσμο και αγοράσετε κάποιο προϊόν\. Για περισσότερες λεπτομέρειες\, |  μπορείτε να μάθετε πώς χρησιμοποιούμε τους συνδέσμους συνεργατών\. |Σας ευχαριστούμε για την υποστήριξη\.|Ακολουθήστε το |  για νέα και ειδήσεις στον χώρο της τεχνολογίας\. Αν ψάχνετε HOT προσφορές και κουπόνια για κινητά και gadgets\, κάντε εγγραφή στο |  στο )$/)`

But also the cosmetic one doesn't still work on mobile. Could you, please, adjust it?

**Reply by freezer2022** [reply, marked as answer]
Try these:


`gizchina.gr##+js(rmnt, #text, /^\s*(?:-----------|Μπορεί να λάβουμε ένα μικρό ποσοστό εάν κάνετε κλικ σε έναν σύνδεσμο και αγοράσετε κάποιο προϊόν\. Για περισσότερες λεπτομέρειες\,|μπορείτε να μάθετε πώς χρησιμοποιούμε τους συνδέσμους συνεργατών\.|Σας ευχαριστούμε για την υποστήριξη\.|Ακολουθήστε το |για νέα και ειδήσεις στον χώρο της τεχνολογίας\. Αν ψάχνετε HOT προσφορές και κουπόνια για κινητά και gadgets\, κάντε εγγραφή στο|στο )\s*$/)`

```adblock
gizchina.gr##.post-content > :is(br + b, br + b + b, br + b + b + b, div + p + br, br:not(p:last-of-type + br))
```
